pkg/controllers/management/rkeworkerupgrader/plan.go

```diff
@@ -119,14 +125,38 @@ func (uh *upgradeHandler) workerPlan(node *v3.Node, cluster *v3.Cluster) (*rkety
 					return np, err
 				}
 			}
+			if cluster.Annotations[clusterprovisioner.RkeRestoreAnnotation] == "true" {
+				// This is for the node agent to be able to detect changes in the node plan for kubelet,
+				// and therefore to recreate the kubelet container on worker nodes after RKE restores an etcd snapshot.
+				// The value is not meant to represent the node version, it is chosen because it increases linearly.
+				newEnvVar := fmt.Sprintf("%s=%d", restoreKey, cluster.Status.NodeVersion)
+				logrus.Debugf("[workerplan] adding/updating env var [%s] on node [%s]", newEnvVar, node.Name)
+				np.Processes[kubeletProcess] = AddEnvVarToProcess(np.Processes[kubeletProcess], newEnvVar)
+			} else {
+				// There is bug that node agent does not restart the kube container if an env var is dropped,
```

Comment:
If we do fix the code around node agent restarting kubelet when env var is dropped, it is going to unnecessarily cause additional kubelet restart after restore process is completed. So I don't think that helps us much. Could you reframe the comment to indicate that you replace it back with existing env var value so agent doesn't keep restarting kubelet container? 

Technically this approach still has the problem of agent restarting kubelet containers multiple times if agent crashes or node goes down during the plan apply, so it'll try to reconcile the desired plan but in that case node is anyway not in "normal" state so that's ok.

Reply:
The comment is updated.